docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,582 @@
+
+## Arguments
+
+`otelcol.processor.attributes` doesn't support any arguments and is configured fully
+through inner blocks.
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`otelcol.processor.attributes`:
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+output | [output][] | Configures where to send received telemetry data. | yes
+action | [action][] | Actions to take on the attributes of incoming metrics/logs/traces. | no
+include | [include/exclude] | Filter for data to be included to this processor's actions. | no
+include > regexp | [regexp] | Regex cache settings. | no
+include > attribute | [attribute] | A list of attributes to match against. | no
+include > resource | [resource] | A list of items to match the resources against. | no
+include > library | [library] | A list of items to match the implementation library against. | no
+include > log_severity_number | [library] | How to match against a log record's SeverityNumber, if defined. | no
+exclude | [include/exclude] | Filter for data to be excluded from this processor's actions | no
+exclude > regexp | [regexp] | Regex cache settings. | no
+exclude > attribute | [attribute] | A list of attributes to match against. | no
+exclude > resource | [resource] | A list of items to match the resources against. | no
+exclude > library | [library] | A list of items to match the implementation library against. | no
+exclude > log_severity_number | [library] | How to match against a log record's SeverityNumber, if defined. | no
+
+[output]: #output-block
+[action]: #action-block
+[include/exclude]: #include/exclude-blocks
+[regexp]: #regexp-block
+[attribute]: #attribute-block
+[resource]: #resource-block
+[library]: #library-block
+[log_severity_number]: #log_severity_number-block
+
+### action block
+
+The `action` block configures how to modify the span, log or metric.
+
+The following attributes are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`key` | `string` | The attribute that the action relates to. |  | yes
+`action` | `string` | The type of action to be performed. |  | yes
+`value` | `any` | The value to populate for the key. |  | no
+`pattern` | `string` | A regex pattern. | `""` | no
+`from_attribute` | `string` | The attribute from the input data to use to populate the attribute value. | `""` | no
+`from_context` | `string` | The context value to use to populate the attribute value.  | `""` | no
+`converted_type` | `string` | The type to convert the attribute value to. | `""` | no
+
+The type of `value` could be a number, a string or a boolean.
+
+The supported values for `action` are:
+
+* `insert`: Inserts a new attribute in input data where the key does not already exist.
+
+    * `key` is required. It specifies the attribute to act upon.
+    * One of `value`, `from_attribute` or `from_context` is required.
+    * `action = "insert"` is required.
```

Comment:
`action = "insert"` being required when `action` is `insert` is a little confusing to me. Can we drop this? 

Ditto with the other instances of this below (`action = "upsert"`, etc)